File: Problem002/sol.cpp
#include <iostream>

using namespace std;

int main() {
	int i = 0, j = 1;
	int current = 0;
	int result = 0;
	
	while(current < 4000000) {
		current = i + j;
		
		if (current % 2 == 0) {
			result += current;
		}
		
		i = j;
		j = current;
	}
	
	cout << result << endl;
	return 0;
}

File: Problem002/sol.cs
using System;

public class sol {
	public static void Main() {
		int previous = 1;
		int result = 0;
		int x = 0;
		
		for (int i = 1; i <= 4000000; i += previous) {
			if(x > 0) {
				previous = i - previous;
			}
			
			x++;
			
			if (i % 2 == 0) {
				result += i;
			}
		}
		
		Console.WriteLine(result);
	}
}
